projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -115,7 +117,13 @@ def run_step(context: JobContext, step: Step) -> None:
         out: HookCallResult
         out = yield
 
-        exception = out.get_result().exception
+        result: StepResult
+        if out.excinfo:
+            result = context.core_context.state.get(CommonStoreKeys.STEP_RESULTS)[-1]
+        else:
+            result = out.get_result()
```

Comment:
Replacing one line with 5 is clearly regression/worse than before. I don't think it's a good idea.